app/controllers/concerns/circuit_breaker.rb

```diff
@@ -0,0 +1,19 @@
+module CircuitBreaker
+  class NotAvailablePathMissingError < RuntimeError; end
+
+  extend ActiveSupport::Concern
+
+  included do
+    rescue_from GetIntoTeachingApiClient::CircuitBrokenError, with: :redirect_to_not_available
+  end
+
+protected
+
+  def redirect_to_not_available
+    redirect_to(not_available_path)
+  end
+
+  def not_available_path
+    raise NotAvailablePathMissingError
```

Comment:
Should this be returning a path rather than re-raising?

Reply:
I think this was meant as a way to ensure any class that extends the CircuitBreaker concern implements `not_available_path`

Reply:
Ah, fair enough - maybe a tiny explanation in the error message could help here:

```ruby
raise NotAvailablePathMissingError, "#not_available_path hasn't been overridden"
```